packages/advanced-logic/src/extensions/payment-network/address-based.ts

```diff
@@ -141,7 +142,42 @@ export default abstract class AddressBasedPaymentNetwork<
     };
   }
 
-  protected abstract isValidAddress(_address: string, _networkName?: string): boolean;
+  protected isValidAddress(address: string, networkName?: string): boolean {
+    if (networkName) {
+      return this.isValidAddressForNetwork(address, networkName);
+    }
+    return this.supportedNetworks.some((network) =>
+      this.isValidAddressForNetwork(address, network),
+    );
+  }
+
+  protected isValidAddressForNetwork(address: string, network: string): boolean {
+    switch (this.supportedCurrencyType) {
+      case RequestLogicTypes.CURRENCY.BTC:
+        return this.isValidAddressForSymbolAndNetwork(
+          address,
+          network === 'testnet' ? 'BTC-testnet' : 'BTC',
+          network,
+        );
+      case RequestLogicTypes.CURRENCY.ETH:
+      case RequestLogicTypes.CURRENCY.ERC20:
+        return this.isValidAddressForSymbolAndNetwork(address, 'ETH', 'mainnet');
+      default:
+        throw new Error(
+          `Default implementation of isValidAddressForNetwork() does not support currency type ${this.supportedCurrencyType}. Please override this method if needed.`,
+        );
+    }
+  }
+
+  protected isValidAddressForSymbolAndNetwork(
+    address: string,
+    symbol: string,
+    network: string,
+  ): boolean {
+    const currencyManager = CurrencyManager.getDefault();
+    const currency = currencyManager.from(symbol, network);
+    return CurrencyManager.validateAddress(address, currency);
```

Comment:
Could this method be called for a token that is not in the default Currency Manager, or only for ETH/BTC/NEAR.... ?
Using the default currency manager may be a problem here.

Reply:
It cannot as it is designed because it's a protected method of `AddressBasedPaymentNetwork` that is only used by the class and its descendants. It is only called with the `ETH`, `BTC`, and `NEAR` tokens.

I hesitated a lot between placing this method as a public helper in the `CurrencyManager` where it would just chain a `from` and a `validateAddress`
```ts
isValidAddressForSymbolAndNetwork(
    address: string,
    symbol: string,
    network: string,
): boolean {
    const currency = currencyManager.from(symbol, network);
    return this.validateAddress(address, currency);
}
```
; or as a protected method in the `AddressBasedPaymentNetwork`.

I choose to place it in the `AddressBasedPaymentNetwork` as it is only used here for now, in order to prevent adding another public API to `CurrencyManager` that we have to maintain. But this choice is debatable, tell me what you prefer, I can move it easily.

Reply:
I think it's OK 👍

Reply:
PS. In any case (even if we move it to the `CurrencyManager`), the `AddressBasedPaymentNetwork` would still be using the default currency manager.